app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java

```diff
@@ -0,0 +1,199 @@
+package com.kabouzeid.gramophone.views;
+
+import android.content.Context;
+import android.graphics.Rect;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.widget.FrameLayout;
+import android.widget.HorizontalScrollView;
+import android.widget.TextView;
+
+import com.kabouzeid.gramophone.R;
+
+/**
+ * @author Lincoln (theduffmaster)
+ * 
+ * A custom FrameLayout view that intercepts touch events and decides whether to consume them or
+ * pass on the touch events to its children.
+ */
+public class TouchInterceptFrameLayout extends FrameLayout {
+
+    private static final int MAX_CLICK_DISTANCE = 5;
+    private static final int MAX_VERTICAL_DISTANCE = 5;
+
+    private Context c;
+
+    private String TAG;
+    private String NULL_VIEWS_EXCEPTION_MESSAGE;
+    
+    private HorizontalScrollView scrollView;
+    private TextView textView;
+    private Rect scrollViewRect = new Rect();
+    private float startX;
+    private float startY;
+    private boolean isTap;
+    private String songTruncated;
+    private String song;
+
+    public TouchInterceptFrameLayout(@NonNull Context context) {
+        super(context);
+        c= context;
+        TAG = c.getResources().getString(R.string.TOUCH_INTERCEPT_FRAME_LAYOUT_TAG);
+        NULL_VIEWS_EXCEPTION_MESSAGE = c.getResources().getString(R.string.NULL_VIEWS_EXCEPTION_MESSAGE);
+    }
+
+    public TouchInterceptFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        c= context;
+        TAG = c.getResources().getString(R.string.TOUCH_INTERCEPT_FRAME_LAYOUT_TAG);
+        NULL_VIEWS_EXCEPTION_MESSAGE = c.getResources().getString(R.string.NULL_VIEWS_EXCEPTION_MESSAGE);
+    }
+
+    public TouchInterceptFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        c= context;
+        TAG = c.getResources().getString(R.string.TOUCH_INTERCEPT_FRAME_LAYOUT_TAG);
+        NULL_VIEWS_EXCEPTION_MESSAGE = c.getResources().getString(R.string.NULL_VIEWS_EXCEPTION_MESSAGE);
+    }
+
+    /**
+     * Fetches the HorizontalScrollView contained by this FrameLayout and also gets the
+     * TextView that is contained within that HorizontalScrollView.
+     * Must be called before setTruncateText.
+     * Call this when you are assigning views in your layout.
+     * @param sv The HorizontalScrollView containing text that needs to be scrolled
+     * @param tv The TextView that needs to be scrolled (typically song or album title)
+     */
+    public void setScrollViews(HorizontalScrollView sv, TextView tv) {
```

Comment:
I'm also thinking that there might be a better name for this than `TouchInterceptFrameLayout` since it pretty specifically handles the truncated/scrolling text, but I'm not sure what exactly to call it. I'm also sort of wondering if this could actually be simplified even further such that the `TextView` doesn't even need to be manually added in the XML.